File: JavaScript/Rock_paper_scissor.js
///* "Rock, Paper, Scissors" game implemented in JavaScript:*/
function getUserChoice() {
  let userChoice = prompt("Choose rock, paper, or scissors: ").toLowerCase();
  while (!["rock", "paper", "scissors"].includes(userChoice)) {
    alert("Invalid choice. Please choose rock, paper, or scissors.");
    userChoice = prompt("Choose rock, paper, or scissors: ").toLowerCase();
  }
  return userChoice;
}

function getComputerChoice() {
  const choices = ["rock", "paper", "scissors"];
  const randomIndex = Math.floor(Math.random() * 3);
  return choices[randomIndex];
}

function determineWinner(userChoice, computerChoice) {
  if (userChoice === computerChoice) {
    return "It's a tie!";
  }
  if (
    (userChoice === "rock" && computerChoice === "scissors") ||
    (userChoice === "scissors" && computerChoice === "paper") ||
    (userChoice === "paper" && computerChoice === "rock")
  ) {
    return "You win!";
  } else {
    return "Computer wins!";
  }
}

function playGame() {
  const userChoice = getUserChoice();
  const computerChoice = getComputerChoice();

  console.log(`You chose ${userChoice}.`);
  console.log(`The computer chose ${computerChoice}.`);

  const result = determineWinner(userChoice, computerChoice);
  console.log(result);
}

console.log("Welcome to Rock, Paper, Scissors!");
while (true) {
  playGame();
  const playAgain = prompt("Do you want to play again? (yes/no): ").toLowerCase();
  if (playAgain !== "yes") {
    console.log("Thanks for playing! Goodbye!");
    break;
  }
}

File: Python/Rock_paper_scissor.py
#simple Python game called "Rock, Paper, Scissors"
import random

def get_user_choice():
    user_choice = input("Choose rock, paper, or scissors: ").lower()
    while user_choice not in ['rock', 'paper', 'scissors']:
        print("Invalid choice. Please choose rock, paper, or scissors.")
        user_choice = input("Choose rock, paper, or scissors: ").lower()
    return user_choice

def get_computer_choice():
    return random.choice(['rock', 'paper', 'scissors'])

def determine_winner(user_choice, computer_choice):
    if user_choice == computer_choice:
        return "It's a tie!"
    if (user_choice == 'rock' and computer_choice == 'scissors') or (user_choice == 'scissors' and computer_choice == 'paper') or (user_choice == 'paper' and computer_choice == 'rock'):
        return "You win!"
    else:
        return "Computer wins!"

def play_game():
    user_choice = get_user_choice()
    computer_choice = get_computer_choice()

    print(f"You chose {user_choice}.")
    print(f"The computer chose {computer_choice}.")

    result = determine_winner(user_choice, computer_choice)
    print(result)

if __name__ == "__main__":
    print("Welcome to Rock, Paper, Scissors!")
    while True:
        play_game()
        play_again = input("Do you want to play again? (yes/no): ").lower()
        if play_again != 'yes':
            print("Thanks for playing! Goodbye!")
            break
